ixwebsocket/IXWebSocketPoll.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ *  IXWebSocketPoll.cpp
+ *  Author: Korchynskyi Dmytro
+ *  Copyright (c) 2017-2018 Machine Zone, Inc. All rights reserved.
+ */
+
+#include "IXWebSocketPoll.h"
+
+namespace ix
+{
+
+    WebSocketPoll::WebSocketPoll(WebSocket* websocket)
+    {
+        bindWebsocket(websocket);
+    }
+
+    WebSocketPoll::~WebSocketPoll()
+    {
+        if (!_messages.empty())
+        {
+            // not handled all messages
+        }
+
+        bindWebsocket(nullptr);
+    }
+
+    void WebSocketPoll::bindWebsocket(WebSocket * websocket)
+    {
+        if (_websocket != websocket)
+        {
+            // unbind old
+            if (_websocket)
+            {
+                _websocket->setOnMessageCallback(nullptr);
+            }
+
+            _websocket = websocket;
+
+            // bind new
+            if (_websocket)
+            {
+                _websocket->setOnMessageCallback([this](
+                    WebSocketMessageType type,
+                    const std::string& str,
+                    size_t wireSize,
+                    const WebSocketErrorInfo& errorInfo,
+                    const WebSocketOpenInfo& openInfo,
+                    const WebSocketCloseInfo& closeInfo)
+                {
+                    MessageDataPtr message(new Message());
+
+                    message->type      = type;
+                    message->str       = str;
+                    message->wireSize  = wireSize;
+                    message->errorInfo = errorInfo;
+                    message->openInfo  = openInfo;
+                    message->closeInfo = closeInfo;
+
+                    _messagesMutex.lock();
```

Comment:
I'd like to avoid explicit mutex lock and unlock, and instead use lock_guard which works better with exceptions, and take less code (no need for the explicit unlock)

Reply:
Ok